bgpd/bgp_nht.c

```diff
@@ -745,8 +745,16 @@ static void evaluate_paths(struct bgp_nexthop_cache *bnc)
 			bnc_is_valid_nexthop =
 				bgp_isvalid_labeled_nexthop(bnc) ? 1 : 0;
 		} else {
-			bnc_is_valid_nexthop =
-				bgp_isvalid_nexthop(bnc) ? 1 : 0;
+			if (bgp_update_martian_nexthop(
+				    bnc->bgp, afi, safi, path->type,
+				    path->sub_type, path->attr, rn)) {
+				zlog_debug(
+					"Prefix %pRN, ignoring path due to martian or self-next-hop",
```

Comment:
wouldn't this be a duplicate logging? we have already:
```
if (BGP_DEBUG(nht, NHT))
			zlog_debug("%s: prefix %pRN (vrf %s) %svalid", __func__,
				   rn, bgp_path->name,
				   (bnc_is_valid_nexthop ? "" : "not "));
```
maybe re-setting here bnc_is_valid_nexthop to 0 is enough? or just do nothing?

Reply:
Thanks, @ton31337. At this point, the bnc_is_valid_nexthop is 0 anyway. So, effectively we're doing nothing. However, I added the additional log message just to flag the reason more specifically. I thought it might be useful for the admins to know why the route is not resolved. Please let me know your thoughts on this.

Reply:
If we agree on this, we still need to guard zlog_debug with `if (BGP_DEBUG(nht, NHT))` and reformat the message like:
`"%s: prefix %pRN (vrf %s), ignoring path due to martian or self-next-hop", __func__, ...` or so to be consistent with others. I would like to hear others about this as well.

Reply:
Thanks, @ton31337. I've made the recommended changes. Kindly have a look.